nucypher/config/base.py

```diff
@@ -606,20 +621,23 @@ def produce(self, **overrides) -> CHARACTER_CLASS:
         return character
 
     @classmethod
-    def assemble(cls, filepath: str = None, **overrides) -> dict:
+    def assemble(cls, filepath: Path = None, **overrides) -> dict:
         """
         Warning: This method allows mutation and may result in an inconsistent configuration.
         """
         payload = cls._read_configuration_file(filepath=filepath)
         node_storage = cls.load_node_storage(storage_payload=payload['node_storage'],
                                              federated_only=payload['federated_only'])
-        domain = payload['domain']
         max_gas_price = payload.get('max_gas_price')  # gwei
         if max_gas_price:
             max_gas_price = Decimal(max_gas_price)
 
         # Assemble
-        payload.update(dict(node_storage=node_storage, domain=domain, max_gas_price=max_gas_price))
+        payload.update(dict(node_storage=node_storage, max_gas_price=max_gas_price))
+        for key in ('keyring_root', 'filepath', 'config_root', 'registry_filepath'):
```

Comment:
I worry a little about code maintenance here i.e. ensuring that this set tuple of keys is kept up to date. Unsure what can be done though...